src/spike/spike.service.ts

```diff
@@ -1,116 +1,43 @@
-     */
-    public async getToken (): Promise<string> {
-        if (this.token && this.checkTokenValidity()) {
-            return this.token.token;
-        }
-        log(Severity.INFO, 'attempting to obtain token from redis', 'getToken');
-        let kartoffelToken:string|null = await this.getAsyncRedis('kartoffel:token');
-        let updateMongo: boolean = false;
-        let tokenExp: number = 0;
-        if (kartoffelToken) {
-            // TODO: Token verification?
-            log(Severity.INFO, 'successfully obtained token', 'getToken');
-            return kartoffelToken;
-        }
+    public async getToken() {
 
-        // If the token is not in redis - either because we didn't save it yet or it has been expired
-        log(Severity.INFO, 'failed to get token from redis. checking mongo.', 'getToken');
-        let tokenObject : IToken | null | void;
-        // Check if the token exists in mongo
-        try {
-            tokenObject = await tokenModel.findOne({ id: tokenID }).exec();
-        } catch (err) {
-            log(Severity.ERROR, err.toString(), 'getToken: mongo findOne');
-        }
-        try {
-            if (!tokenObject || tokenObject.expireAt.getTime() < Date.now()) {
-                log(Severity.INFO, 'failed to get token from mongo. fetching from kartoffel.', 'getToken');
-                const spikeRes: SpikeToken = await this.renewToken();
-                tokenExp = parseInt(spikeRes.expires_in, 10);
-                this.redis.set('kartoffel:token', spikeRes.access_token, 'EX', tokenExp);
-                kartoffelToken = spikeRes.access_token;
-                updateMongo = true;
-            } else {
-                this.token = tokenObject;
-                kartoffelToken = tokenObject.token;
-            }
-        } catch (err) {
-            throw new ServerError(`Error in receiving token: ${err}`);
-        }
+        const client = await new spike_proto.Spike('spike-service:8080', grpc.credentials.createInsecure());
 
-        // Add the new token to mongo
-        if (updateMongo) {
-            const newToken : IToken = {
-                id: tokenID,
-                token: kartoffelToken,
-                expireAt: new Date(Date.now() + tokenExp * Second),
-            };
-            try {
-                this.token = newToken;
-                await tokenModel.findOneAndUpdate({ id: tokenID }, newToken, { upsert: true }).exec();
-            } catch (err) {
-                log(Severity.ERROR, err.toString(), 'getToken: mongo findOneAndUpdate');
-            }
-        }
-
-        // TODO: Token verification?
-        log(Severity.INFO, 'successfully obtained token', 'getToken');
-        return kartoffelToken;
+        return this.getSpikeToken(client);
     }
 
-    private checkTokenValidity(): boolean {
-        return (!!this.token && this.token.expireAt.getTime() > Date.now());
+    private getSpikeToken(client: any) {
```

Comment:
Add the return type in the method statement. should something like 'Promise<THE_TYPE>'